website/pages/docs/platform/k8s/helm/examples.mdx

```diff
@@ -55,23 +55,159 @@ ui:
+        [req]
+        req_extensions = v3_req
+        distinguished_name = req_distinguished_name
+        [req_distinguished_name]
+        [ v3_req ]
+        basicConstraints = CA:FALSE
+        keyUsage = nonRepudiation, digitalSignature, keyEncipherment
+        extendedKeyUsage = serverAuth
+        subjectAltName = @alt_names
+        [alt_names]
+        DNS.1 = ${SERVICE}
+        DNS.2 = ${SERVICE}.${NAMESPACE}
+        DNS.3 = ${SERVICE}.${NAMESPACE}.svc
+        DNS.4 = ${SERVICE}.${NAMESPACE}.svc.cluster.local
+        IP.1 = 127.0.0.1
+        EOF
+        ```
+
+    2. Create a CSR.
+
+        ```bash
+        openssl req -new -key ${TMPDIR}/vault.key -subj "/CN=${SERVICE}.${NAMESPACE}.svc" -out ${TMPDIR}/server.csr -config ${TMPDIR}/csr.conf
+        ```
+
+3. Create the certificate
+
+    1. Create a file `${TMPDIR}/csr.yaml` with the following contents:
+
+        ```
+        $ export CSR_NAME=vault-csr
+        $ cat <<EOF >${TMPDIR}/csr.yaml
+        apiVersion: certificates.k8s.io/v1beta1
+        kind: CertificateSigningRequest
+        metadata:
+          name: ${CSR_NAME}
+        spec:
+          groups:
+          - system:authenticated
+          request: $(cat ${TMPDIR}/server.csr | base64 | tr -d '\n')
+          usages:
+          - digital signature
+          - key encipherment
+          - server auth
+        EOF
+        ```
+        -> `CSR_NAME` can be any name you want. It's the name of the CSR as seen by Kubernetes
+
+    2. Send the CSR to Kubernetes.
+
+        ```bash
+        kubectl create -f ${TMPDIR}/csr.yaml
+        ```
+        -> If this process is automated, you may need to wait to ensure the CSR has been received and stored:
+        `kubectl get csr ${CSR_NAME}`
+
+    3. Approve the CSR in Kubernetes.
+
+        ```bash
+        kubectl certificate approve ${CSR_NAME}
+        ```
+
+### 2. Store key, cert, and Kubernetes CA into Kubernetes secrets store
+
+1. Retrieve the certificate.
+
+    ```bash
+    serverCert=$(kubectl get csr ${CSR_NAME} -o jsonpath='{.status.certificate}')
+    ```
+   -> If this process is automated, you may need to wait to ensure the certificate has been created.
+   If it hasn't, this will return an empty string.
+
+2. Write the certificate out to a file.
+
+    ```bash
+    echo "${serverCert}" | openssl base64 -d -A -out ${TMPDIR}/vault.crt
+    ```
+
+3. Retrieve Kubernetes CA.
+
+    ```bash
+    kubectl config view --raw --minify --flatten -o jsonpath='{.clusters[].cluster.certificate-authority-data}' | base64 -D > ${TMPDIR}/vault.ca
+    ```
+
+3. Store the key, cert, and Kubernetes CA into Kubernetes secrets.
+
+    ```bash
+    kubectl create secret generic ${SECRET_NAME} \
+            --namespace ${NAMESPACE} \
+            --from-file=vault.key=${TMPDIR}/vault.key \
+            --from-file=vault.crt=${TMPDIR}/vault.crt \
+            --from-file=vault.ca=${TMPDIR}/vault.ca
+    ```
+
+
+
+## 3. Helm Configuration
```

Comment:
Should this be an h3 to match the other headings in this section?

```suggestion
### 3. Helm Configuration
```

Reply:
:+1: